compiler+runtime/src/cpp/main.cpp

```diff
@@ -124,27 +121,40 @@ namespace jank
       dynamic_call(__rt_ctx->in_ns_var->deref(), make_box<obj::symbol>(opts.target_module));
     }
 
-    /* By default, RL will do tab completion for files. We disable that here. */
-    rl_bind_key('\t', rl_insert);
+    auto const get_prompt([&](native_persistent_string const &suffix) {
+      return __rt_ctx->current_ns()->name->to_code_string() + suffix;
+    });
+
+    /* By default we are placed in clojure.core ns as of now.
+     * TODO: Set default ns to `user` when we are dropped in that ns.*/
+    llvm::LineEditor le("jank", ".jank-repl-history");
+    le.setPrompt(get_prompt("=> "));
+    native_transient_string input{};
 
     /* TODO: Completion. */
     /* TODO: Syntax highlighting. */
     /* TODO: Multi-line input. */
-    while(auto const buf = readline("> "))
+    while(std::optional<native_persistent_string> buf = le.readLine())
     {
-      native_transient_string line{ buf };
+      native_transient_string line(*buf);
       boost::trim(line);
-      if(line.empty())
+
+      if(line.ends_with("\\"))
       {
+        input.append(line.substr(0, line.size() - 1));
+        le.setPrompt(get_prompt("=>... "));
         continue;
       }
 
-      /* TODO: Persist history to disk, á la .lein-repl-history. */
-      /* History is persisted for this session only. */
-      add_history(line.data());
+      input += line;
+      if(input == "(exit)")
+      {
+        break;
+      }
```

Comment:
Why add this? Clojure doesn't do it. Is there any other precedent?

It prevents the user from defining a fn named exit and calling it. Normal `^d` does the trick.

Reply:
I see. I've seen this being used in a REPL. I can't recall where.

But it makes sense to not include it as you mentioned.